zone_test.go

```diff
@@ -2,15 +2,155 @@ package cloudflare
 
 import (
 	"context"
+	"crypto/md5"   // for generating IDs
+	"encoding/hex" // for generating IDs
+	"encoding/json"
 	"fmt"
 	"net/http"
 	"net/url"
+	"strconv"
 	"testing"
 	"time"
 
 	"github.com/stretchr/testify/assert"
 )
 
+// mockID returns a hex string of length 32, suitable for all kinds of IDs
+// used in the Cloudflare API.
+func mockID(seed string) string {
+	arr := md5.Sum([]byte(seed))
+	return hex.EncodeToString(arr[:])
+}
+
+func mustParseTime(s string) time.Time {
+	t, err := time.Parse(time.RFC3339Nano, s)
+	if err != nil {
+		panic(err)
+	}
+	return t
+}
+
+func mockZone(i int) *Zone {
+	zoneName := fmt.Sprintf("%d.example.com", i)
+	ownerName := "Test Account"
+
+	return &Zone{
+		ID:      mockID(zoneName),
+		Name:    zoneName,
+		DevMode: 0,
+		OriginalNS: []string{
+			"linda.ns.cloudflare.com",
+			"merlin.ns.cloudflare.com",
+		},
+		OriginalRegistrar: "cloudflare, inc. (id: 1910)",
+		OriginalDNSHost:   "",
+		CreatedOn:         mustParseTime("2021-07-28T05:06:20.736244Z"),
+		ModifiedOn:        mustParseTime("2021-07-28T05:06:20.736244Z"),
+		NameServers: []string{
+			"abby.ns.cloudflare.com",
+			"noel.ns.cloudflare.com",
+		},
+		Owner: Owner{
+			ID:        mockID(ownerName),
+			Email:     "",
+			Name:      ownerName,
+			OwnerType: "organization",
+		},
+		Permissions: []string{
+			"#access:read",
+			"#analytics:read",
+			"#auditlogs:read",
+			"#billing:read",
+			"#dns_records:read",
+			"#lb:read",
+			"#legal:read",
+			"#logs:read",
+			"#member:read",
+			"#organization:read",
+			"#ssl:read",
+			"#stream:read",
+			"#subscription:read",
+			"#waf:read",
+			"#webhooks:read",
+			"#worker:read",
+			"#zone:read",
+			"#zone_settings:read",
+		},
+		Plan: ZonePlan{
+			ZonePlanCommon: ZonePlanCommon{
+				ID:       "0feeeeeeeeeeeeeeeeeeeeeeeeeeeeee",
+				Name:     "Free Website",
+				Currency: "USD",
+			},
+			IsSubscribed:      false,
+			CanSubscribe:      false,
+			LegacyID:          "free",
+			LegacyDiscount:    false,
+			ExternallyManaged: false,
+		},
+		PlanPending: ZonePlan{
+			ZonePlanCommon: ZonePlanCommon{
+				ID: "",
+			},
+			IsSubscribed:      false,
+			CanSubscribe:      false,
+			LegacyID:          "",
+			LegacyDiscount:    false,
+			ExternallyManaged: false,
+		},
+		Status: "active",
+		Paused: false,
+		Type:   "full",
+		Host: struct {
+			Name    string
+			Website string
+		}{
+			Name:    "",
+			Website: "",
+		},
+		VanityNS:    nil,
+		Betas:       nil,
+		DeactReason: "",
+		Meta: ZoneMeta{
+			PageRuleQuota:     3,
+			WildcardProxiable: false,
+			PhishingDetected:  false,
+		},
+		Account: Account{
+			ID:   mockID(ownerName),
+			Name: ownerName,
+		},
+		VerificationKey: "",
+	}
+}
+
+func mockResultInfo(total, page, start, count int) *ResultInfo {
+	return &ResultInfo{
+		Page:       page,
+		PerPage:    50,
+		TotalPages: (total + 49) / 50,
+		Count:      count,
+		Total:      total,
+	}
+}
```

Comment:
Ah! I should have avoided hardcoding the number `50` in it unless all zone-related API uses 50 as the maximum number of items per page. :confused: Let me make one final change.